zulip_bots/zulip_bots/lib.py

```diff
@@ -65,7 +66,8 @@ def put(self, key, value):
 
     def get(self, key):
         # type: (Text) -> Text
-        return self.demarshal(self.state_[key])
+        # Return a copy of the entry to prevent a bot accidentally modifying self.state_ on its own.
```

Comment:
It seems like we could reorder the commits to have the JSON change happen first, so that we don't need to do the deepcopy here.

Reply:
The problem I see is when bots change the default encoding; The new encoding must be guaranteed to not return any reference to the old object. Maybe we can make an additional check for this?

Reply:
Hmm, but then again I can't see a case where that could occur, except for the original `lamba obj: obj`, which wouldn't work anymore anyway.